### PR TITLE
Correct Segment GA schema; default to userId if null

### PR DESCRIPTION
**Pull request by jordanh**
Nick – I'd love to get this PR in the next release. Last release [we made a change](https://github.com/ParabolInc/parabol/pull/4117) to try and fix a data irregularity with Google Analytics. When we reviewed how this fix had been performing with the Segment Solutions Architect, he realized he gave us the wrong data shape to implement. This PR fixes this so we can try for reals.

### CODE REVIEW

**Comment by jordanh** [comment]
Addresses #4116

File: packages/server/utils/segmentIo.ts [hunk 1]
```diff
@@ -24,27 +24,25 @@ segmentIo.track = async (options) => {
   return (segmentIo as any)._track({
     userId,
     event,
-    context: {
```
Comment (jordanh) [inline packages/server/utils/segmentIo.ts:27]: Note: this context block is removed

File: packages/server/utils/segmentIo.ts [hunk 2]
```diff
@@ -24,27 +24,25 @@ segmentIo.track = async (options) => {
...
+     * in this way. When we would looked at our data, events coming from our app would
+     * be tallied as "direct" rather than acquired through a known channel (e.g.
+     * "Organic Search"), making it look like all these new users just dropped out of the
+     * sky.
+     *
+     * To work around this, we specify which a UUID Segment will consistently hash
+     * between our marketing site and app domains. We do this by passing in
+     * additional Segment context, as seen here:
+     */
+    integrations: {
```
Comment (jordanh) [inline packages/server/utils/segmentIo.ts:44]: now, the `integrations` block is a peer to `properties`

**Review by nickoferrall — APPROVED** [review]
LGTM @jordanh! Is there anything that you would like me to test?

**Comment by jordanh** [comment]
@nickoferrall no tests on this one are needed. I had validated locally, and, it can't possibly get more broken than it already is :)

**Pull request merged**